x-pack/plugins/spaces/server/lib/request_interceptors/on_request_interceptor.test.ts

```diff
@@ -0,0 +1,284 @@
+              path: req.path,
+              basePath: http.getBasePathFor(req),
+              query: req.query,
+            });
+          },
+        },
+      ]);
+    }
+
+    if (routes === 'new-platform') {
+      const router = new Router('/');
+
+      router.get({ path: '/foo', validate: false }, (req: KibanaRequest, h: any) => {
+        return h.ok({ path: req.url.pathname, basePath: http.getBasePathFor(req) });
+      });
+
+      router.get(
+        { path: '/some/path/s/foo/bar', validate: false },
+        (req: KibanaRequest, h: any) => {
+          return h.ok({ path: req.url.pathname, basePath: http.getBasePathFor(req) });
+        }
+      );
+
+      router.get(
+        {
+          path: '/i/love/spaces',
+          validate: schema => {
+            return {
+              query: schema.object({
+                queryParam: schema.string({
+                  defaultValue: 'oh noes, this was not set on the request correctly',
+                }),
+              }),
+            };
+          },
+        },
+        (req: KibanaRequest, h: any) => {
+          return h.ok({
+            path: req.url.pathname,
+            basePath: http.getBasePathFor(req),
+            query: req.query,
+          });
+        }
+      );
+
+      http.registerRouter(router);
+    }
+  }
+
+  describe('requests proxied to the legacy platform', () => {
+    it('handles paths without a space identifier', async () => {
+      const { http } = await root.setup();
+
+      const basePath = '/';
+      const config = ({
+        get: jest.fn().mockReturnValue(basePath),
+      } as unknown) as KibanaConfig;
+
+      initSpacesOnRequestInterceptor({ config, http });
+
+      await root.start();
+
+      initKbnServer(http, 'legacy');
+
+      const path = '/foo';
+
+      await kbnTestServer.request.get(root, path).expect(200, {
+        path,
+        basePath: '', // no base path set for route within the default space
+      });
+    }, 30000);
+
+    it('strips the Space URL Context from the request', async () => {
+      const { http } = await root.setup();
+
+      const basePath = '/';
+      const config = ({
+        get: jest.fn().mockReturnValue(basePath),
+      } as unknown) as KibanaConfig;
+
+      initSpacesOnRequestInterceptor({ config, http });
+
+      await root.start();
+
+      initKbnServer(http, 'legacy');
+
+      const path = '/s/foo-space/foo';
+
+      const resp = await kbnTestServer.request.get(root, path);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.body).toEqual({
+        path: '/foo',
+        basePath: '/s/foo-space',
+      });
+    }, 30000);
+
+    it('ignores space identifiers in the middle of the path', async () => {
+      const { http } = await root.setup();
+
+      const basePath = '/';
+      const config = ({
+        get: jest.fn().mockReturnValue(basePath),
+      } as unknown) as KibanaConfig;
+
+      initSpacesOnRequestInterceptor({ config, http });
```

Comment:
why not register specific router right here?

Reply:
Do you mean instead of having that logic within `initKbnServer`? No reason one way or the other...this was a port of the existing tests, so I didn't change the structure around a whole lot.